### PR TITLE
Remove queued presets from Thermostats

### LINKED ISSUE #34089: Remove QueuedPresets from the Thermostat cluster
This issue tracks changes in the sdk corresponding to the spec issue - https://github.com/CHIP-Specifications/connectedhomeip-spec/pull/9324

### PR DESCRIPTION
**Pull request by nivi-apple**
Fixes: #34089

The PR has skip-protocol-compatibility due to the fact that some of the attributes are not marked provisional in the xml when they should be. The attributes are not implemented yet. However since they are not marked provisional we need to add the skip-protocol-compatibility so API compatibility checker passes. These are the attributes that were removed - 

Thermostat::SetActivePresetRequestRequest has field changes
Bitmap Feature removed entry kQueuedPresetsSupported
Command Thermostat::CancelSetActivePresetRequest was removed
Attribute Thermostat::QueuedPreset has been deleted.

For the Thermostat::QueuedPresetStruct, it was marked provisional but we still saw errors and an issue was filed for that - https://github.com/project-chip/connectedhomeip/issues/34091

### CODE REVIEW

**Comment by andy31415** [comment]
@nivi-apple please add some description in the PR summary of why `skip-protocol-compatibility` is correct for this PR

**Review by kiel-apple — APPROVED** [review]

File: src/app/zap-templates/zcl/data-model/chip/thermostat-cluster.xml
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/data-model/chip/thermostat-cluster.xml:1]: The corresponding entries in Darwin availability.yaml should be removed, right?

**Review by bzbarsky-apple — APPROVED** [review]

**Comment by nivi-apple** [comment]
@andy31415 Please see https://github.com/project-chip/connectedhomeip/pull/34090#issue-2376014935

File: src/app/zap-templates/zcl/data-model/chip/thermostat-cluster.xml [hunk 1]
```diff
@@ -523,10 +513,6 @@ limitations under the License.
       <description>This command is used to notify the server that all edits are done and should be committed.</description>
       <access op="invoke" privilege="manage"/>
     </command>
-    <command source="client" code="0x0A" name="CancelSetActivePresetRequest" optional="true">
```
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/data-model/chip/thermostat-cluster.xml:526]: This is still in availability.yaml.

Comment (nivi-apple) [reply src/app/zap-templates/zcl/data-model/chip/thermostat-cluster.xml:526]: done.

File: src/app/zap-templates/zcl/data-model/chip/thermostat-cluster.xml
Comment (nivi-apple) [reply src/app/zap-templates/zcl/data-model/chip/thermostat-cluster.xml:1]: done.

**Review by andy31415 — APPROVED** [review]

**Pull request merged**